core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/SettingsDownloader.java

```diff
@@ -0,0 +1,165 @@
+    GlobalProperties.Builder builder = GlobalProperties.newBuilder();
+    fetchSettings(serverVersion, null, (k, v) -> true, builder::putProperties);
+    return builder.build();
+  }
+
+  public void fetchProjectSettings(String serverVersion, String moduleKey, GlobalProperties globalProps, ModuleConfiguration.Builder projectConfigurationBuilder) {
+    fetchSettings(serverVersion, moduleKey, (k, v) -> !v.equals(globalProps.getPropertiesMap().get(k)), projectConfigurationBuilder::putProperties);
+  }
+
+  private void fetchSettings(String serverVersion, @Nullable String moduleKey, BiPredicate<String, String> filter, BiConsumer<String, String> consumer) {
+    if (Version.create(serverVersion).compareToIgnoreQualifier(Version.create("6.3")) >= 0) {
+      fetchUsingSettingsWS(moduleKey, consumer);
+    } else {
+      fetchUsingPropertiesWS(moduleKey, filter, consumer);
+    }
+  }
+
+  private void fetchUsingSettingsWS(@Nullable String moduleKey, BiConsumer<String, String> consumer) {
+    String url = API_SETTINGS_PATH;
+    if (moduleKey != null) {
+      url += "?component=" + StringUtils.urlEncode(moduleKey);
+    }
+    WsResponse response = wsClient.get(url);
+    try (InputStream is = response.contentStream()) {
+      ValuesWsResponse values = ValuesWsResponse.parseFrom(is);
+      for (Setting s : values.getSettingsList()) {
+        // Storage optimisation: don't store settings having same value than global settings
+        if (!s.getInherited()) {
+          processSetting(consumer, s);
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to parse properties from: " + response.content(), e);
+    }
+  }
+
+  private void processSetting(BiConsumer<String, String> consumer, Setting s) {
+    switch (s.getValueOneOfCase()) {
+      case VALUE:
+        consumer.accept(s.getKey(), s.getValue());
+        break;
+      case VALUES:
+        consumer.accept(s.getKey(), Joiner.on(',').join(s.getValues().getValuesList()));
+        break;
+      case FIELDVALUES:
+        processPropertySet(s, consumer);
+        break;
+      default:
+        throw new IllegalStateException("Unknow property value for " + s.getKey());
+    }
+  }
+
+  private static void processPropertySet(Setting s, BiConsumer<String, String> consumer) {
+    List<String> ids = new ArrayList<>();
+    int id = 1;
+    for (Value v : s.getFieldValues().getFieldValuesList()) {
+      for (Map.Entry<String, String> entry : v.getValue().entrySet()) {
+        consumer.accept(s.getKey() + "." + id + "." + entry.getKey(), entry.getValue());
+      }
+      ids.add(String.valueOf(id));
+      id++;
+    }
+    consumer.accept(s.getKey(), Joiner.on(',').join(ids));
+  }
+
+  private void fetchUsingPropertiesWS(String moduleKey, BiPredicate<String, String> filter, BiConsumer<String, String> consumer) {
+    String url = API_PROPERTIES_PATH;
+    if (moduleKey != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583)